### PR TITLE
Extract ManageOptionController

### PR DESCRIPTION
**Pull request by JDawes-ScottLogic**
Extract ManageOptionController from ManageController and refactor front end to use it instead.
Added tests for refactor and new controller.

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Api/Models/DBViewModels/ManageOptionUpdateRequest.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            Options = new List<OptionUpdate>();
+        }
+
+        public List<OptionUpdate> Options { get; set; }
+    }
+
+    public class OptionUpdate
+    {
+        public string Name { get; set; }
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Api/Models/DBViewModels/ManageOptionUpdateRequest.cs]: We should mark Name as a required field

**Comment by JDawes-ScottLogic** [comment]
Added Required field.
Investigating whether Except is needed. I added it later, so believe it is, but can't remember why (and the tests pass without it...).

Also found an issue with add/edit dialogs not working properly, which needs correcting.

**Comment by tpkelly** [comment]
Looks fine to me, +1

**Comment by stevenhillcox** [comment]
Let me know when you are happy for this to be merged.

**Pull request merged**